add whisper quadlet, move whisper model-service, update docs (#83)
containers/ai-lab-recipes: audio-to-text/README.md

# Audio to Text Application

  This sample application is a simple recipe to transcribe an audio file.
  This provides a simple recipe to help developers start building out their own custom LLM enabled
  audio-to-text applications. It consists of two main components; the Model Service and the AI Application.

  There are a few options today for local Model Serving, but this recipe will use [`whisper-cpp`](https://github.com/ggerganov/whisper.cpp.git)

>   and their OpenAI compatible Model Service. There is a Containerfile provided that can be used to build this Model Service within the repo,

Reviewer: I don't think this model uses an OpenAI compatible API.
https://platform.openai.com/docs/api-reference/audio/createTranscription?lang=curl

We can probably just say, "and their included model server"

Author: done

  [`model_servers/whispercpp/Containerfile`](/model_servers/whispercpp/Containerfile).

>   Our AI Application will connect to our Model Service via it's OpenAI compatible API.

Reviewer: same note as above

<p align="center">
<img src="../assets/whisper.png" width="70%">
</p>

# Build the Application

In order to build this application we will need a model, a Model Service and an AI Application.  

* [Download a model](#download-a-model)
* [Build the Model Service](#build-the-model-service)
* [Deploy the Model Service](#deploy-the-model-service)
* [Build the AI Application](#build-the-ai-application)
* [Deploy the AI Application](#deploy-the-ai-application)
* [Interact with the AI Application](#interact-with-the-ai-application)
    * [Input audio files](#input-audio-files)

### Download a model

If you are just getting started, we recommend using [ggerganov/whisper.cpp](https://huggingface.co/ggerganov/whisper.cpp).

> This is a well performant mid-sized model with an apache-2.0 license.

Reviewer: ```suggestion
This is a well performant model with an MIT license.
```

### Build the Model Service

The Model Service can be built from the root directory with the following code snippet:

```bash
cd model_servers/whispercpp
podman build -t whispercppserver .
```

### Deploy the Model Service

The local Model Service relies on a volume mount to the localhost to access the model files. You can start your local Model Service using the following podman command:  
```
podman run --rm -it \
        -p 8001:8001 \
        -v Local/path/to/locallm/models:/locallm/models \
        -e MODEL_PATH=models/<model-filename> \
        -e HOST=0.0.0.0 \
        -e PORT=8001 \
        whispercppserver
```

### Build the AI Application

Now that the Model Service is running we want to build and deploy our AI Application. Use the provided Containerfile to build the AI Application
image from the `audio-to-text/` directory.

```bash
cd audio-to-text
podman build -t audio-to-text . -f builds/Containerfile   
```
### Deploy the AI Application

Make sure the Model Service is up and running before starting this container image.
When starting the AI Application container image we need to direct it to the correct `MODEL_SERVICE_ENDPOINT`.

> This could be any appropriately hosted Model Service (running locally or in the cloud) using an OpenAI compatible API.

Reviewer: ```suggestion
This could be any appropriately hosted Model Service (running locally or in the cloud) using a compatible API.
```

The following podman command can be used to run your AI Application:  

```bash
podman run --rm -it -p 8501:8501 -e MODEL_SERVICE_ENDPOINT=http://0.0.0.0:8001/inference audio-to-text 
```

### Interact with the AI Application

Once the streamlit application is up and running, you should be able to access it at `http://localhost:8501`.
From here, you can upload audio files from your local machine and translate the audio files as shown below.

> Everything should now be up an running with the chat application available at [`http://localhost:8501`](http://localhost:8501).

Reviewer: seems duplicative of above
```suggestion

```

By using this recipe and getting this starting point established,

> users should now have an easier time customizing and building their own LLM enabled chatbot applications.   

Reviewer: ```suggestion
users should now have an easier time customizing and building their own LLM enabled applications.   
```
